src/compare.js

```diff
@@ -212,6 +212,25 @@ function compareWithInt(actual, rule, path) {
   if (type !== 'number') {
     throw `Json doesn't have type 'number' at '${path}' but found '${type}'`;
   } else {
+    if ('options' in rule && rule.options !== undefined) {
```

Comment:
Move this new code to a common function and reuse for both `int` and `float` comparisons